trieTools/tokensExporter/main.go

```diff
@@ -239,31 +268,69 @@ func getAllESDTTokens(account vmcommon.AccountHandler, pubKeyConverter core.Pubk
 		// TODO: Try to unmarshal it when the new meta data storage model will be live
 		tokenKey := leaf.Key()
 		lenESDTPrefix := len(esdtPrefix)
-		tokenName := getPrettyTokenName(tokenKey[lenESDTPrefix:])
+		prettyTokenIdentifier, prettyTokenName := getPrettyTokenName(tokenKey[lenESDTPrefix:])
+		tokenID, nonce := common.ExtractTokenIDAndNonceFromTokenStorageKey(tokenKey[lenESDTPrefix:])
+
+		var isTokenOfInterest = false
+
+		for _, tokenOfInterest := range tokensOfInterest {
+			if strings.HasPrefix(prettyTokenIdentifier, tokenOfInterest) {
+				isTokenOfInterest = true
+				break
+			}
+		}
+
+		if !isTokenOfInterest {
+			continue
+		}
+
+		esdtTokenKey := []byte(core.ProtectedKeyPrefix + core.ESDTKeyIdentifier + string(tokenID))
+		esdtToken, _, err := GetESDTNFTTokenOnDestination(userAccount, esdtTokenKey, nonce, systemAccount)
+		if err != nil {
+			log.Warn("cannot get ESDT token", "token name", prettyTokenIdentifier, "error", err)
+			continue
+		}
+
+		item := accountToken{
+			Identifier: prettyTokenIdentifier,
+			Name:       prettyTokenName,
+			Balance:    esdtToken.Value.String(),
+		}
 
-		allESDTs[tokenName] = struct{}{}
+		if esdtToken.TokenMetaData != nil {
+			item.Nonce = esdtToken.TokenMetaData.Nonce
+			item.Attributes = esdtToken.TokenMetaData.Attributes
+			item.Creator = addressConverter.Encode(esdtToken.TokenMetaData.Creator)
+		}
+
+		allTokens = append(allTokens, item)
 	}
 
 	err = common.GetErrorFromChanNonBlocking(iteratorChannels.ErrChan)
 	if err != nil {
 		return nil, err
 	}
 
-	return allESDTs, nil
+	return allTokens, nil
 }
 
-func getPrettyTokenName(tokenName []byte) string {
+func getPrettyTokenName(tokenName []byte) (string, string) {
 	token, nonce := common.ExtractTokenIDAndNonceFromTokenStorageKey(tokenName)
+	rebuiltName := ""
 	if nonce != 0 {
 		tokens := bytes.Split(token, []byte("-"))
 
+		rebuiltName = string(tokens[0]) + "-" + string(tokens[1])
```

Comment:
not sure of the intent of this. The token already was `ABC-010101` and rebuiltName should still be `ABC-010101`